letsencrypt/cli.py

```diff
@@ -1106,8 +1107,47 @@ def parse_args(self):
                                        "'certonly' or 'renew' subcommands")
                 parsed_args.break_my_certs = parsed_args.staging = True
 
+        if parsed_args.csr:
+            self.handle_csr(parsed_args)
+
         return parsed_args
 
+    def handle_csr(self, parsed_args):
+        """
+        Process a --csr flag. This needs to happen early enought that the
+        webroot plugin can know about the calls to _process_domain
+        """
+        try:
+            csr = le_util.CSR(file=parsed_args.csr[0], data=parsed_args.csr[1], form="der")
+            typ = OpenSSL.crypto.FILETYPE_ASN1
+            domains = crypto_util.get_sans_from_csr(csr.data, OpenSSL.crypto.FILETYPE_ASN1)
+        except:
```

Comment:
pylint doesn't like this broad except and needs a pylint ignore if you're sure of it. Same thing with the except at line 1130. I wonder whether this could produce a "parse error" when the CSR file simply can't be read or was wrongly specified? Like if you say `--csr my_crs.pem` instead of `--csr my_csr.pem`?